### ISSUE TITLE
debug js/tsx from installed app

### ISSUE THREAD

**Issue by cwtowns** [microsoft/react-native-windows#7615]
Once an app is installed, how do I debug the tsx/js code?  The VS solution lets me Debug Installed App Package and select the installed package ... but then what?  How do I attach chrome so I can step through the js/tsx and see console statements?  

Edit:  I guess the main question I'm asking is whether or not it is possible to see console.log statements when the app is compiled in release mode.  At a glance it seems like a no.

**Comment by asklar** [comment]
Ideally you debug your app by running the Debug build. The Dev menu is disabled in Release mode, as are other debug-related features (yellow boxes, and the default red box I believe?). An app also doesn't use Chrome to run when built in Release mode. If you disable web debugging (either manually or by running Release mode), then you should still be able to debug the JS code by attaching Visual Studio to the app in "Script mode" (open VS, Debug->Attach->select type=Script). I'm not sure if this will be able to resolve minified JS, Andrew will know.


Meta-point for triage: we don't talk about the e2e debug experience in our docs but we should; there are Windows-specific pieces (e.g. how to use VS, does flipper work, JS engine choice) as well as RN concepts that the main RN website doesn't go into enough detail either (web debugging vs. direct debugging).

**Comment by chrisglein** [comment]
> Edit: I guess the main question I'm asking is whether or not it is possible to see console.log statements when the app is compiled in release mode. At a glance it seems like a no.

Correct, you won't get those when the bundle is in release.
You can turn the debug menu back on in release.

> Meta-point for triage: we don't talk about the e2e debug experience in our docs but we should; there are Windows-specific pieces (e.g. how to use VS, does flipper work, JS engine choice) as well as RN concepts that the main RN website doesn't go into enough detail either (web debugging vs. direct debugging).

Yes, we need more documentation for this. Creating an issue for that [here](https://github.com/microsoft/react-native-windows-samples/issues/426).

If you need more answers ping this issue again.

**Comment by cwtowns** [comment]
Well I narrowed down what my problem is and am looking for guidance.  I'm trying to use [uuidv4](https://www.npmjs.com/package/uuidv4) which depends on the crypto package.  This works fine in debug but not in release.  In order to track down uuidv4 as the problem in release mode, I added a native module that writes to file via Serilog, and then updated my front end to log to that instead of the console.  This lead me to the problematic line which I wrapped in a try / catch and then wrote the caught error out via the native module I added:  

`Error: crypto.getRandomValues() not supported. See https://github.com/uuidjs/uuid#getrandomvalues-not-supported`

This is a long way to go to track down an error message.  Is there a better way?  I don't see uncaught errors being logged in VS when I run from there in Release.  Where would they be?  

@asklar If I run in script mode, how would I set the break points?  The js/tsx isn't in the solution.
